lib/pure/streams.nim

```diff
@@ -116,173 +292,447 @@ when not defined(js):
+    strm.close()
   if readData(s, addr(result), sizeof(result)) != 1: result = '\0'
 
 proc peekChar*(s: Stream): char =
-  ## peeks a char from the stream `s`. Raises `IOError` if an error occurred.
+  ## Peeks a char from the stream `s`. Raises `IOError` if an error occurred.
   ## Returns '\\0' as an EOF marker.
+  runnableExamples:
+    var strm = newStringStream("12\n3")
+    doAssert strm.peekChar() == '1'
+    doAssert strm.peekChar() == '1'
+    discard strm.readAll()
+    doAssert strm.peekChar() == '\x00'
+    strm.close()
   if peekData(s, addr(result), sizeof(result)) != 1: result = '\0'
 
 proc readBool*(s: Stream): bool =
-  ## reads a bool from the stream `s`. Raises `IOError` if an error occurred.
+  ## Reads a bool from the stream `s`. Raises `IOError` if an error occurred.
+  runnableExamples:
+    var strm = newStringStream("12")
+    doAssert strm.readBool()
+    doAssert strm.readBool()
+    ## strm.readBool() --> raise IOError
+    strm.close()
   read(s, result)
 
 proc peekBool*(s: Stream): bool =
-  ## peeks a bool from the stream `s`. Raises `IOError` if an error occurred.
+  ## Peeks a bool from the stream `s`. Raises `IOError` if an error occurred.
+  runnableExamples:
+    var strm = newStringStream("12")
+    doAssert strm.peekBool()
+    doAssert strm.peekBool()
+    doAssert strm.peekBool()
+    strm.close()
   peek(s, result)
 
 proc readInt8*(s: Stream): int8 =
-  ## reads an int8 from the stream `s`. Raises `IOError` if an error occurred.
+  ## Reads an int8 from the stream `s`. Raises `IOError` if an error occurred.
+  runnableExamples:
+    var strm = newStringStream("12")
+    doAssert strm.readInt8() == 49
+    doAssert strm.readInt8() == 50
+    ## strm.readInt8() --> raise IOError
+    strm.close()
   read(s, result)
 
 proc peekInt8*(s: Stream): int8 =
-  ## peeks an int8 from the stream `s`. Raises `IOError` if an error occurred.
+  ## Peeks an int8 from the stream `s`. Raises `IOError` if an error occurred.
+  runnableExamples:
+    var strm = newStringStream("12")
+    doAssert strm.peekInt8() == 49
+    doAssert strm.peekInt8() == 49
+    doAssert strm.peekInt8() == 49
+    strm.close()
   peek(s, result)
 
 proc readInt16*(s: Stream): int16 =
-  ## reads an int16 from the stream `s`. Raises `IOError` if an error occurred.
+  ## Reads an int16 from the stream `s`. Raises `IOError` if an error occurred.
+  runnableExamples:
+    var strm = newStringStream("001020")
+    doAssert strm.readInt16() == 12336
+    doAssert strm.readInt16() == 12337
+    doAssert strm.readInt16() == 12338
+    ## strm.readInt16() --> raise IOError
+    strm.close()
   read(s, result)
 
 proc peekInt16*(s: Stream): int16 =
-  ## peeks an int16 from the stream `s`. Raises `IOError` if an error occurred.
+  ## Peeks an int16 from the stream `s`. Raises `IOError` if an error occurred.
+  runnableExamples:
+    var strm = newStringStream("001020")
+    doAssert strm.peekInt16() == 12336
+    doAssert strm.peekInt16() == 12336
```

Comment:
Same as above.

Maybe these examples would work better if you `peek` then `read`, then again `peek` and `read`? Just like you did below for `proc peekLine`.